### DISCUSSION TITLE
Automatic documentation

### DISCUSSION THREAD

**Discussion by Guillermogsjc** [art049/odmantic#251] [Ideas]
# Feature request

### Context

Anyone here knows  Open API standard documentation, and how using pydantic and fastapi, the `openapi.json` is automatically generated and it can be injected into a openapi.json renderer to obtain a HTML doc UI and a standard way to document the API models.

### Idea

The idea/question is, there exists any standard for document databases, similar to Open API for APIs, where we can take the created `odmantic` models, and generate automatically document databases model documentation at once? 

In case of not existing nothing similar, how would you create something like that using `odmantic`?

**Comment by art049** [comment]
Hey, well I'm not sure such a thing exists, unfortunately. I know mongo supports [JSON Schema Validation](https://www.mongodb.com/docs/manual/core/schema-validation/specify-json-schema/#std-label-schema-validation-json) so it's technically possible to turn an ODMantic model to some database side validation logic. It's planned but not a priority at all for now.

If you're interested I'd be glad to help you to implement this.
